lib/eth/contract.rb

```diff
@@ -37,27 +37,58 @@ def initialize(name, bin, abi)
       @constructor_inputs, @functions, @events = parse_abi(abi)
     end
 
-    # Creates a contract wrapper.
+    # Creates a contract wrapper from a Solidity file.
     #
     # @param file [String] solidity file path.
-    # @param bin [String] contract bin string.
+    # @param contract_index [Number] specify contract.
+    # @return [Eth::Contract::Object] Returns the class of the smart contract.
+    # @raise [ArgumentError] if the file path is empty or no contracts were compiled.
+    def self.from_file(file: nil, contract_index: 0)
+      if File.exist?(file.to_s)
+        contracts = Eth::Contract::Initializer.new(file).build_all
+        raise ArgumentError, "No contracts compiled." if contracts.empty?
+        contract = contracts[contract_index].class_object.new
+      else
+        raise ArgumentError, "Cannot find the contract at #{file.to_s}!"
+      end
+      return contract
```

Comment:
```suggestion
     raise ArgumentError, "Cannot find the contract at #{file.to_s}!" if !File.exist?(file.to_s)
     contracts = Eth::Contract::Initializer.new(file).build_all
     raise ArgumentError, "No contracts compiled." if contracts.empty?
     
     contracts[contract_index].class_object.new
```

I kind of feel like the Initializer class should be in charge of its own validation, so if the class is used in multiple places, this validation doesn't have to be repeated. 

If that change is made, then this method could be shortened to:

```suggestion
      contracts = Eth::Contract::Initializer.new(file).build_all
      contracts[contract_index].class_object.new
```